Fixed the demo server comparing the wrong timestamps

farcaller opened the pull request:
The lease must be further in time than the current moment.

Commit:
Fixed the demo server comparing the wrong timestamps

(the lease must be further in time than the current moment)

krolaw reviewed (changes requested):
Hang on, the check is just to find out if a given IP is available.  Thus, if the lease has expired (or the current lease for that slot matches the chaddr), then it is available.  But you've still found a bug.  It should be:
Some(x) => x.0 == *chaddr || Instant::now().gt(&x.1),

farcaller commented:
Ah, I see what you meant there originally.

farcaller committed:
Fixed the availability detection code

The IP address is considered available if either holds true:

- it's bound to the same MAC
- its lease had expired
